### PR TITLE
Default to mongosh when connecting to MongoDB. (#8472)

### LINKED ISSUE #8472: Support `mongosh` when connecting to a MongoDB
### What

`tsh db connect` currently uses `mongo` CLI client when connecting to a MongoDB database. `mongo` is obsolete now and is going to be eventually replaced by `mongosh` to `tsh db connect` should support using it too.

### How

`tsh db connect` should try to use `mongosh` and fallback to `mongo` if it's not available.

### Notes

Customers say `mongosh` also handles CA somewhat differently, might require additional tweaks:

> mongosh handles the ca differently too. from my testing, i couldn't get mongosh to use the trusted ca in my system keychain (macos). would be good to add that as something we should try handling too.

### PR DESCRIPTION
**Pull request by Tener**
Makes the `mongosh` default MongoDB command line client to look for. If it isn't available, we fallback to legacy `mongo` client. For `mongosh` we use updated set of flags (e.g. `--tls` instead of `--ssl`), which are available since MongoDB 4.2.

The docs and tests are updated accordingly.

The branch is based on (approved for merging) #9409 and incorporates changes discussed in #9451. 

Fixes #8472.

### CODE REVIEW

**Comment by smallinsky** [comment]
@Tener Could you rebase so only MongoSH changes will be present without MariaDB https://github.com/gravitational/teleport/pull/9409 changes ?

File: tool/tsh/db.go [hunk 1]
```diff
@@ -560,78 +597,204 @@ func getConnectCommand(cf *CLIConf, tc *client.TeleportClient, profile *client.P
...
+}
+
+func (c *cliCommandBuilder) getMySQLOracleCommand() *exec.Cmd {
+	args := c.getMySQLCommonCmdOpts()
+
+	// defaults-group-suffix must be first.
+	groupSuffix := []string{fmt.Sprintf("--defaults-group-suffix=_%v-%v", c.tc.SiteName, c.db.ServiceName)}
+	args = append(groupSuffix, args...)
+
+	// override the ssl-mode from a config file is --insecure flag is provided to 'tsh db connect'.
```
Comment (zmb3) [inline tool/tsh/db.go:692]:
```suggestion
	// override the ssl-mode from a config file if --insecure flag is provided to 'tsh db connect'.
```

**Review by zmb3 — APPROVED** [review]
Approved, though note one small typo in a comment.

**Comment by r0mant** [comment]
@Tener I think you need to rebase, looks like this includes changes from @jakule's PR.

**Comment by Tener** [comment]
Making a new PR based on code from master.

**Pull request closed without merging**